core/src/Platforms/iOS/TokenCacheAccessor.cs

```diff
@@ -32,90 +32,248 @@
+            var account = cacheKey.HomeAccountId + CacheKeyDelimiter + 
+                cacheKey.Environment;
+
+            var service = cacheKey.CredentialType + CacheKeyDelimiter +
+                          cacheKey.ClientId + CacheKeyDelimiter +
+                          cacheKey.TenantId + CacheKeyDelimiter +
+                          cacheKey.Scopes;
+
+            var type = (int)CredentialAttrType.AccessToken;
+
+            Remove(account, service, type);
         }
 
-        public void DeleteIdToken(string cacheKey)
+        public void DeleteRefreshToken(MsalRefreshTokenCacheKey cacheKey)
         {
-            Remove(cacheKey, IdTokenServiceId);
+            var account = cacheKey.HomeAccountId + CacheKeyDelimiter +
+                          cacheKey.Environment;
+
+            var service = cacheKey.CredentialType + CacheKeyDelimiter +
+                          cacheKey.ClientId + CacheKeyDelimiter +
+                          "" + CacheKeyDelimiter;
+
+            var type = (int)CredentialAttrType.RefreshToken;
+
+            Remove(account, service, type);
         }
 
-        public void DeleteAccount(string cacheKey)
+        public void DeleteIdToken(MsalIdTokenCacheKey cacheKey)
         {
-            Remove(cacheKey, AccountServiceId);
+            var account = cacheKey.HomeAccountId + CacheKeyDelimiter +
+                          cacheKey.Environment;
+
+            var service = cacheKey.CredentialType + CacheKeyDelimiter +
+                          cacheKey.ClientId + CacheKeyDelimiter +
+                          cacheKey.TenantId + CacheKeyDelimiter;
+
+            var type = (int)CredentialAttrType.IdToken;
+
+            Remove(account, service, type);
         }
 
+        public void DeleteAccount(MsalAccountCacheKey cacheKey)
+        {
+            var account = cacheKey.HomeAccountId + CacheKeyDelimiter +
+                          cacheKey.Environment;
+
+            var service = cacheKey.TenantId;
+
+            var type = AuthorityTypeToAttrType[AuthorityType.MSSTS.ToString()];
+
+            Remove(account, service, type);
+        }
+        
+
         public ICollection<string> GetAllAccessTokensAsString()
         {
-            return GetValues(AccessTokenServiceId);
+            return GetValues((int)CredentialAttrType.AccessToken);
         }
 
         public ICollection<string> GetAllRefreshTokensAsString()
         {
-            return GetValues(RefreshTokenServiceId);
+            return GetValues((int)CredentialAttrType.RefreshToken);
         }
 
         public ICollection<string> GetAllIdTokensAsString()
         {
-            return GetValues(IdTokenServiceId);
+            return GetValues((int)CredentialAttrType.IdToken);
         }
 
         public ICollection<string> GetAllAccountsAsString()
         {
-            return GetValues(AccountServiceId);
+            return GetValues(AuthorityTypeToAttrType[AuthorityType.MSSTS.ToString()]);
         }
-
+        /*
```

Comment:
dead code

Reply:
unifed cache is not finalized yet, code cleanup will be done later

Reply:
I may be misunderstanding the utility of the "dev" branch. My understanding is that dev branch should be stable because it is where we run tests and it acts as a de-facto release branch. As developers, we do not control the release schedule, and we could be asked to ship at any time. So having clean code in dev is pretty important. 

Maybe you should push all your changes to a feature branch and then merge everything with dev  when everything is complete?

Reply:
Context: For now we use dev as an all features branch for common core. adalv3dev is the real dev branch for adal. When cache and web views are done we will complete an FTP to validate the state of the entire branch (which we have never tested fully e2e since the refactoring's into core started). For this particular branch this approach is expected until we reach the above mentioned milestones. After that point it will switch to a true dev branch for our unified libraries. Hope this helps explain the reasoning behind.